Library/Homebrew/dependency.rb

```diff
@@ -4,15 +4,22 @@
 class Dependency
   include Dependable
 
-  attr_reader :name, :tags, :env_proc, :option_name
+  WARNED_CROSS_TAP_DEPENDENCY = []
+
+  attr_reader :name, :tags, :env_proc, :option_name, :owner_tap
 
   DEFAULT_ENV_PROC = proc {}
 
-  def initialize(name, tags = [], env_proc = DEFAULT_ENV_PROC, option_name = name)
+  def initialize(name, tags = [], owner_tap = nil, env_proc = DEFAULT_ENV_PROC, option_name = name)
     @name = name
     @tags = tags
     @env_proc = env_proc
     @option_name = option_name
+    if (not owner_tap.nil?) and owner_tap.to_s =~ HOMEBREW_TAP_ARGS_REGEX
+      @owner_tap = "#{$1}/#{$3.gsub(/^homebrew-/, "")}"
+    else
+      @owner_tap = nil
+    end
```

Comment:
In fact, you can just make it `@owner_tap = owner_tap` here. The reason is `Formulary.factory` can accept ref likes `homebrew/homebrew-foo/bar`.